### PR TITLE
DEP-3813 Added audit events to controller load events.

### PR DESCRIPTION
**Pull request by jameshforster**
# DEP-3813

## Checklist PR Raiser
 - [ ]  I've ensured code coverage has not decreased
 - [ ]  I've dealt with any new compilation warnings
 - [ ]  I've ensured the team's coding standards have been met (TBC)?
 - [ ]  If relevant, I've created corresponding app-config-XXX changes?
 - [ ]  If I've created new test data, I've ensured there is no real person's data (see https://confluence.tools.tax.service.gov.uk/display/CD/Test+Data+in+the+Open) 
 - [ ]  I've considered the impact my changes have on the UI/Journey tests (please do not break them)
-  [ ]  If relevant, I've added links to associated PRs

## Checklist PR Reviewer
 - [ ]  I've checked to ensure all relevant unit tests have been written
 - [ ]  I've checked to ensure the team's coding standards have been met
 - [ ]  I've executed the acceptance test pack locally to ensure there are no functional regressions (Performance tests where relevant)
 - [ ]  If I merge I will ensure I use squash and merge

### CODE REVIEW

File: test/controllers/CiapiController/CiapiControllerSpec.scala [hunk 1]
```diff
@@ -16,37 +16,62 @@
...
+import views.html.CUIViews._
 import views.html.pages.helpers.AppBuilderSpecBase
 
 class CiapiControllerSpec
-  extends AppBuilderSpecBase with Matchers with AnyWordSpecLike {
-
-  private val controller = app.injector.instanceOf[CiapiController]
+  extends AppBuilderSpecBase with Matchers with AnyWordSpecLike with MockAuditService {
+
+  def controller = new CiapiController(
```
Comment (machinesleet) [inline test/controllers/CiapiController/CiapiControllerSpec.scala:37]: is there a reason this is a def and not a val?

**Comment by machinesleet** [comment]
The new trade tariff page has gone in to the main branch since this PR, so the audit helper and tests will need updating

File: app/audit/AuditHelper.scala [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.audit.AuditExtensions
+import uk.gov.hmrc.play.audit.http.connector.AuditConnector
+import uk.gov.hmrc.play.audit.model.DataEvent
+
+import scala.concurrent.ExecutionContext
+
+@Singleton
+class AuditHelper @Inject()(appConfig: AppConfig, auditConnector: AuditConnector) {
+  def audit(auditModel: AuditModel, path: Option[String] = None)(implicit hc: HeaderCarrier, request: Request[_], ec: ExecutionContext): Unit = {
```
Comment (machinesleet) [inline app/audit/AuditHelper.scala:32]: Could you explain why we're passing in the path argument to this function, when we're only using the request path for the DataEvent tags field?

Comment (jameshforster) [reply app/audit/AuditHelper.scala:32]: That would be legacy from copying the functionality from another service, I don't think we use it so I will remove it.

File: test/controllers/CiapiController/CiapiControllerSpec.scala (hunk 1 above)
Comment (jameshforster) [reply test/controllers/CiapiController/CiapiControllerSpec.scala:37]: Typically due to safety. Running a def (and generating a new controller for each test) helps prevent cross-test contamination. It's not explicitly required, I can remove it, just a practice I picked up working on teams with complex mocking logic where there was a bigger risk of that happening.

File: app/audit/AuditHelper.scala (hunk 2 above)
Comment (machinesleet) [reply app/audit/AuditHelper.scala:32]: OK, but at the moment are we using the optional path at all? Could we not remove this argument and just have the path from the request

File: test/controllers/CiapiController/CiapiControllerSpec.scala (hunk 1 above)
Comment (machinesleet) [reply test/controllers/CiapiController/CiapiControllerSpec.scala:37]: I wonder if we specifically want to have one instance of the controller in this case, whilst having multiple tests. Wouldn't this be a closer approximation of the behaviour when running the application?

**Review by machinesleet — APPROVED** [review]

**Pull request merged**